resources/lib/addon/parser.py

```diff
@@ -42,30 +62,30 @@ def try_type(value, output=None):
         return try_float(value)
 
 
-def try_decode(string, encoding='utf-8', errors=None):
+def try_decode(string, encoding='utf-8', errors='strict'):
     """helper to decode strings for PY 2 """
     if sys.version_info.major == 3:
         return string
     try:
-        return string.decode(encoding, errors) if errors else string.decode(encoding)
+        return string.decode(encoding, errors)
     except Exception:
         return string
 
 
-def try_encode(string, encoding='utf-8'):
+def try_encode(string, encoding='utf-8', errors='strict'):
     """helper to encode strings for PY 2 """
     if sys.version_info.major == 3:
         return string
     try:
-        return string.encode(encoding)
+        return string.encode(encoding, errors)
     except Exception:
         return string
 
 
 def parse_paramstring(paramstring):
     """ helper to assist with difference in urllib modules in PY2/3 """
     params = dict()
-    paramstring = paramstring.replace('&amp;', '&')  # Just in case xml string
+    paramstring = try_encode(paramstring).replace('&amp;', '&')  # Just in case xml string
```

Comment:
Is this a typo? Wouldn't you want _decode_ for `str.replace()` and `str.split()` ? Also, pretty sure we want to return decoded params.

Reply:
not a typo, it's to utf-8 encode the string for the calls to unquote_plus

Reply:
but after reviewing, these shims aren't needed... i swore unquote and unquote_plus threw errors when handed unicode values on py2, but alas they do not

Reply:
> not a typo, it's to utf-8 encode the string for the calls to unquote_plus

In PY2 `encode()` transforms a unicode string `u'foobar'` into bytes `b'foobar'`. So in this method you are expecting the paramstring to be unicode. However, elsewhere in the container router you removed decoding from the paramstring so you will be getting bytes here no unicode.

I'd think that you'd want to decode the paramstring to unicode as early as possible (i.e. in container router) and then you don't need to encode/decode again because you're already doing try_encode in the shims.